iina/Extensions.swift

```diff
@@ -316,3 +316,13 @@ extension Data {
     return fileUrl
   }
 }
+
+
+extension CharacterSet {
+  static let urlAllowed = CharacterSet.urlHostAllowed
```

Comment:
This is *technically* wrong but probably the easiest way to do it. I'd add a comment here for posterity explaining that we should be splitting up the URL and encoding each part separately.